src/arm/phyboard.c

```diff
@@ -0,0 +1,672 @@
+    if (uart3_enabled == 1) {
+        strncpy(b->pins[35].name, "X_UART3_RX", MRAA_PIN_NAME_SIZE);
+        b->pins[35].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 1 };
+    } else {
+        strncpy(b->pins[35].name, "X_GPIO2_18", MRAA_PIN_NAME_SIZE);
+        b->pins[35].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 1 };
+    }
+    b->pins[35].gpio.pinmap = (2*32 + 18);
+    b->pins[35].gpio.parent_id = 0;
+    b->pins[35].gpio.mux_total = 0;
+    b->pins[35].uart.mux_total = 0;
+
+    /*  PIN-Number "36" with name "X_UART3_TX_GPIO2_19" */
+    if (uart3_enabled == 1) {
+        strncpy(b->pins[36].name, "X_UART3_TX", MRAA_PIN_NAME_SIZE);
+        b->pins[36].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 1 };
+    } else {
+        strncpy(b->pins[36].name, "X_GPIO2_19", MRAA_PIN_NAME_SIZE);
+        b->pins[36].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 1 };
+    }
+    b->pins[36].gpio.pinmap = (2*32 + 19);
+    b->pins[36].gpio.parent_id = 0;
+    b->pins[36].gpio.mux_total = 0;
+    b->pins[36].uart.mux_total = 0;
+
+    /* PIN-Number "37" with name "X_INTR1_GPIO0_20" */
+    strncpy(b->pins[37].name, "X_GPIO0_20", MRAA_PIN_NAME_SIZE);
+    b->pins[37].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 0 };
+    b->pins[37].gpio.pinmap = (0*32 + 20);
+    b->pins[37].gpio.parent_id = 0;
+    b->pins[37].gpio.mux_total = 0;
+
+    strncpy(b->pins[38].name, "X_GPIO0_7", MRAA_PIN_NAME_SIZE);
+    b->pins[38].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 0 };
+    b->pins[38].gpio.pinmap = (0*32 + 7);
+    b->pins[38].gpio.parent_id = 0;
+    b->pins[38].gpio.mux_total = 0;
+
+    strncpy(b->pins[39].name, "X_AM335_EXT_WAKEUP", MRAA_PIN_NAME_SIZE);
+    b->pins[39].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 0, 0 };
+
+    strncpy(b->pins[40].name, "X_INT_RTCn", MRAA_PIN_NAME_SIZE);
+    b->pins[40].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 0, 0 };
+
+    strncpy(b->pins[41].name, "GND", MRAA_PIN_NAME_SIZE);
+    b->pins[41].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 0, 0 };
+
+    /* PIN-Number "42" with name "X_GPIO3_7_nPMOD_PW RFAIL" */
+    strncpy(b->pins[42].name, "X_GPIO3_7", MRAA_PIN_NAME_SIZE);
+    b->pins[42].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 0 };
+    b->pins[42].gpio.pinmap = (3*32 + 7);
+    b->pins[42].gpio.parent_id = 0;
+    b->pins[42].gpio.mux_total = 0;
+
+    strncpy(b->pins[43].name, "nRESET_IN", MRAA_PIN_NAME_SIZE);
+    b->pins[43].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 0, 0 };
+
+    strncpy(b->pins[44].name, "X_GPIO1_31", MRAA_PIN_NAME_SIZE);
+    b->pins[44].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 0 };
+    b->pins[44].gpio.pinmap = (1*32 + 31);
+    b->pins[44].gpio.parent_id = 0;
+    b->pins[44].gpio.mux_total = 0;
+
+    strncpy(b->pins[45].name, "X_AM335_NMIn", MRAA_PIN_NAME_SIZE);
```

Comment:
`MRAA_PIN_NAME_SIZE` is 12 and `b->pins[45].name` is allocated to 12 symbols, that means the `strncpy()` will not copy the null-terminator of the string and it won't fit anyway - please consider shortening the name.